core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -405,9 +405,7 @@ class SparkContext(config: SparkConf) extends Logging {
 
     _eventLogDir =
       if (isEventLogEnabled) {
-        val unresolvedDir = conf.get("spark.eventLog.dir", EventLoggingListener.DEFAULT_LOG_DIR)
-          .stripSuffix("/")
-        Some(Utils.resolveURI(unresolvedDir))
```

Comment:
i suggest to use `new Path(path).toURI()` instead `new URI(path)` since `new URI(path)` not support space in path. 
It is not necessary to use encode if we use `new Path(path).toURI()`